google/protobuf/cpp_features.proto

```diff
@@ -0,0 +1 @@
+syntax = "proto3";
```

Comment:
What's the purpose of these dummies?  It seems like this could conflict with the actual versions.  Also, it doesn't actually define any custom features, so anyone referencing those would have problems

Reply:
The purpose I had in mind was that an import of these files may be present in a schema originating in another language or toolchain, so having dummies prevents "file not found" errors in pbjs. Left them blank because the definitions within are not of relevance to protobuf.js anyway. But now that you say it, given that the dummies might leak out to other toolchains, it seems like a good idea to include their full contents.

Reply:
Yea, also usually people would import them *and* reference the features defined in them.  So I think you'd still hit errors from missing options definitions?

OTOH copying them verbatim here seems like a good way to get version skew issues.  How do you deal with descriptor.proto?  We've been modifying that pretty rapidly in the last few years

Reply:
The missing definitions aren't an issue for protobuf.js itself. It currently parses any options literally without looking up schemas. Hence the historic comment in the parser on how properties are assumed. Similarly, descriptor.proto [is optional](https://github.com/protobufjs/protobuf.js/tree/master/ext/descriptor) and only updated sporadically, as it is not needed internally for protobuf.js to function. Perhaps a data point that descriptor.proto hasn't leaked out so far.